Lib/pathlib.py

```diff
@@ -281,9 +281,14 @@ def _parse_parts(cls, parts):
         if altsep:
             path = path.replace(altsep, sep)
         drv, root, rel = cls._flavour.splitroot(path)
-        if drv.startswith(sep):
-            # pathlib assumes that UNC paths always have a root.
-            root = sep
+        if drv and not root and not drv.endswith(sep):
```

Comment:
Maybe exclude relative drive paths from the start, such as "C:spam".

```suggestion
        if not root and drv.startswith(sep) and not drv.endswith(sep):
```

Reply:
Good shout